src/jdk.incubator.jextract/share/classes/jdk/incubator/jextract/tool/resources/RuntimeHelper.java.template

```diff
@@ -77,6 +77,10 @@ public class RuntimeHelper {
         }
     }
 
+    public static void freeUpcallStub(MemoryAddress addr) {
+        ABI.freeUpcallStub(addr);
```

Comment:
Perhaps we could register the returned MemoryAddress with a cleaner and then let the GC pick it up and call the `freeUpcallStub` ? Note that either way we're unsafe - e.g. a client might call this `freeUpcallStub` method while some native code is accessing it (which causes a crash). Or the memory address might become unrecheable while some native code is accessing the upcall. So, either way, some discipline is required.

What I like more about the cleaner approach is that, like with libraries, code is typically longer-lived than allocations, so, IMHO it makes sense to leave that one to the GC, so that the user code doesn't have to worry about deallocating that.